src/components/WalletUnlock/LedgerNano.tsx

```diff
@@ -155,8 +163,8 @@ class LedgerNanoSDecryptClass extends PureComponent<Props & INetworkContext, Sta
     this.reset();
   };
 
-  private handleUnlock = (address: TAddress, index: number) => {
-    this.props.onUnlock(WalletService.init({ address, dPath: this.state.dPath.value, index }));
+  private handleUnlock = async (address: TAddress, index: number) => {
+    this.props.onUnlock(await WalletService.init({ address, dPath: this.state.dPath, index }));
```

Comment:
I don't see a way this ever throws, but I will add a try-catch

Reply:
It may not ever throw an error _now_, but if we change the behaviour of `WalletService` in the future, it's possible that it can throw errors. In general, we should try to catch any promise rejections.